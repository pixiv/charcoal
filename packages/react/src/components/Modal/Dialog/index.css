.charcoal-modal-dialog {
  margin: auto;
  position: relative;
  height: fit-content;
  width: 440px;

  background-color: var(--charcoal-color-container-default);
  border-radius: 24px;
}

.charcoal-modal-dialog[data-size='S'] {
  width: 336px;
}

.charcoal-modal-dialog[data-size='M'] {
  width: 440px;
}

.charcoal-modal-dialog[data-size='L'] {
  width: 648px;
}

@media (max-width: 743px) {
  .charcoal-modal-dialog {
    max-width: 440px;
    width: calc(100% - 48px);
  }

  .charcoal-modal-dialog[data-bottom-sheet='true'],
  .charcoal-modal-dialog[data-bottom-sheet='full'] {
    max-width: unset;
    width: 100%;
    border-radius: 0;
    margin: auto 0 0 0;
  }

  .charcoal-modal-dialog[data-bottom-sheet='full'] {
    min-height: 100%;
  }
}

.charcoal-modal-dialog:focus {
  outline: none;
}
