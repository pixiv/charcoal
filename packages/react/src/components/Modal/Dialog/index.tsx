import { forwardRef } from 'react'
import * as React from 'react'
import { useDialog } from '@react-aria/dialog'
import { useForwardedRef } from '../../../_lib/useForwardedRef'
import { Size, BottomSheet } from '..'

import './index.css'
import { useClassNames } from '../../../_lib/useClassNames'

export const Dialog: React.ForwardRefExoticComponent<Omit<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "ref"> & {
  size: Size
  bottomSheet: BottomSheet
} & React.RefAttributes<HTMLDivElement>> = forwardRef<
  HTMLDivElement,
  React.ComponentPropsWithoutRef<'div'> & {
    size: Size
    bottomSheet: BottomSheet
  }
>(function Dialog({ size, bottomSheet, className, ...props }, forwardRef) {
  const ref = useForwardedRef(forwardRef)
  const { dialogProps } = useDialog(
    {
      role: 'dialog',
    },
    ref
  )

  const classNames = useClassNames('charcoal-modal-dialog', className)
  return (
    <div
      className={classNames}
      role={dialogProps.role}
      data-bottom-sheet={bottomSheet}
      tabIndex={dialogProps.tabIndex}
      aria-labelledby={dialogProps['aria-labelledby']}
      onBlur={dialogProps.onBlur}
      data-size={size}
      ref={ref}
      {...props}
    />
  )
})
