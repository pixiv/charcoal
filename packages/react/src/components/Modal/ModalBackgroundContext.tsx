import * as React from 'react'

/**
 * ModalBackgroundのElementが入ったコンテキスト
 */
export const ModalBackgroundContext: React.Context<HTMLElement | null> = React.createContext<HTMLElement | null>(
  null
)
