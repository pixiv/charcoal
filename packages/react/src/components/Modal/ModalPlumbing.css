.charcoal-modal-header-root {
  height: 64px;
  display: grid;
  align-content: center;
  justify-content: center;
}

@media (max-width: 743px) {
  .charcoal-modal-header-root[data-bottom-sheet='true'],
  .charcoal-modal-header-root[data-bottom-sheet='full'] {
    height: 48px;
  }
}

.charcoal-modal-header-title {
  color: var(--charcoal-color-text-default);
  font-size: 16px;
  line-height: 24px;
  font-weight: bold;
  display: flow-root;
}

.charcoal-modal-align {
  padding-left: 16px;
  padding-right: 16px;
}

.charcoal-modal-body {
  padding-bottom: 40px;
  color: var(--charcoal-color-text-default);
}

.charcoal-modal-buttons {
  display: grid;
  grid-auto-flow: row;
  grid-row-gap: 8px;

  padding-top: 16px;
  padding-left: 16px;
  padding-right: 16px;
}
