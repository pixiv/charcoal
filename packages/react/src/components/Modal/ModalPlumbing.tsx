import React from 'react'
import { ModalTitle } from '.'
import styled from 'styled-components'
import { theme } from '../../styled'

export function ModalHeader() {
  return (
    <ModalHeaderRoot>
      <StyledModalTitle />
    </ModalHeaderRoot>
  )
}

const ModalHeaderRoot = styled.div`
  height: 64px;
  display: grid;
  align-content: center;
  justify-content: center;
`

const StyledModalTitle = styled(ModalTitle)`
  ${theme((o) => [o.font.text1, o.typography(16).bold])}
`

export const ModalAlign = styled.div`
  ${theme((o) => [o.padding.horizontal(16)])}
`

export const ModalBody = styled.div`
  ${theme((o) => [o.padding.bottom(40)])}
`

export const ModalButtons = styled.div`
  display: grid;
  grid-auto-flow: row;
  grid-row-gap: 8px;

  ${theme((o) => [o.padding.horizontal(16).top(16)])}
`
