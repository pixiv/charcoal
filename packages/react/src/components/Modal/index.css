.charcoal-modal-background {
  overflow: auto;
  display: flex;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  justify-content: center;
  padding: 40px 0;
  box-sizing: border-box;

  background-color: var(--charcoal-color-container-neutral-default);
}

@media (max-width: 743px) {
  .charcoal-modal-background[data-bottom-sheet='true'],
  .charcoal-modal-background[data-bottom-sheet='full'] {
    padding: 0;
  }
}

.charcoal-modal-close-button {
  position: absolute;
  top: 8px;
  right: 8px;

  color: var(--charcoal-color-text-tertiary-default);
  transition: 0.2s color;
}

.charcoal-modal-title {
  margin: 0;
  font-weight: inherit;
  font-size: inherit;
}
