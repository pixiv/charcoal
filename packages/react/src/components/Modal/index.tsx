import React, { useContext, useRef } from 'react'
import {
  AriaModalOverlayProps,
  Overlay,
  useModalOverlay,
  useOverlay,
} from '@react-aria/overlays'
import styled, { css, useTheme } from 'styled-components'
import { theme } from '../../styled'
import { FocusScope } from '@react-aria/focus'
import { useDialog } from '@react-aria/dialog'
import { AriaDialogProps } from '@react-types/dialog'
import { columnSystem, COLUMN_UNIT, GUTTER_UNIT } from '@charcoal-ui/foundation'
import { unreachable } from '../../_lib'
import { maxWidth } from '@charcoal-ui/utils'
import { useMedia } from '@charcoal-ui/styled'
import { animated, useTransition, easings } from 'react-spring'
import Button, { ButtonProps } from '../Button'
import IconButton from '../IconButton'

export type ModalProps = AriaModalOverlayProps &
  AriaDialogProps & {
    children: React.ReactNode
    zIndex?: number
    title: string
    size?: 'S' | 'M' | 'L'
    bottomSheet?: boolean | 'full'
    isOpen: boolean
    onClose: () => void

    /**
     * https://github.com/adobe/react-spectrum/issues/3787
     * Next.jsで使用する際に発生するエラーの一時的な回避策でdocument.bodyを指定する必要がある
     */
    portalContainer?: HTMLElement
  }

const DEFAULT_Z_INDEX = 10

/**
 * モーダルコンポーネント。
 *
 * @example アプリケーションルートで `<OverlayProvider>` ないし `<CharcoalProvider>` で囲った上で利用する
 * ```tsx
 * import {
 *   OverlayProvider,
 *   Modal,
 *   ModalHeader,
 *   ModalBody,
 *   ModalButtons
 * } from '@charcoal-ui/react'
 *
 * <OverlayProvider>
 *   <App>
 *     <Modal isOpen={state.isOpen} onClose={() => state.close()} isDismissable>
 *       <ModalHeader />
 *       <ModalBody>...</ModalBody>
 *       <ModalButtons>...</ModalButtons>
 *     </Modal>
 *   </App>
 * </OverlayProvider>
 * ```
 */
export default function Modal({
  children,
  zIndex = DEFAULT_Z_INDEX,
  portalContainer,
  ...props
}: ModalProps) {
  const {
    title,
    size = 'M',
    bottomSheet = false,
    isDismissable,
    onClose,
    isOpen = false,
  } = props

  const ref = useRef<HTMLDivElement>(null)
  const { overlayProps, underlayProps } = useOverlay(props, ref)

  const { modalProps } = useModalOverlay(
    props,
    {
      close: onClose,
      isOpen: isOpen,
      // these props are not used actually.
      // https://github.com/adobe/react-spectrum/blob/df14e3fb129b94b310f0397a701b83f006b51dfe/packages/%40react-aria/overlays/src/useModalOverlay.ts
      open: () => {
        // nope
      },
      setOpen: () => {
        // nope
      },
      toggle: () => {
        // nope
      },
    },
    ref
  )

  const { dialogProps, titleProps } = useDialog(props, ref)

  const theme = useTheme()
  const isMobile = useMedia(maxWidth(theme.breakpoint.screen1)) ?? false
  const transitionEnabled = isMobile && bottomSheet !== false
  const transition = useTransition(isOpen, {
    from: {
      transform: 'translateY(100%)',
      backgroundColor: 'rgba(0, 0, 0, 0)',
    },
    enter: {
      transform: 'translateY(0%)',
      backgroundColor: 'rgba(0, 0, 0, 0.4)',
    },
    leave: {
      transform: 'translateY(100%)',
      backgroundColor: 'rgba(0, 0, 0, 0)',
    },
    config: transitionEnabled
      ? { duration: 400, easing: easings.easeOutQuart }
      : { duration: 0 },
  })
  const showDismiss = !isMobile || bottomSheet !== true

  return transition(
    ({ backgroundColor, transform }, item) =>
      item && (
        <Overlay portalContainer={portalContainer}>
          <ModalBackground
            zIndex={zIndex}
            {...underlayProps}
            style={transitionEnabled ? { backgroundColor } : {}}
          >
            <FocusScope contain restoreFocus autoFocus>
              <ModalDialog
                ref={ref}
                {...overlayProps}
                {...modalProps}
                {...dialogProps}
                style={transitionEnabled ? { transform } : {}}
                size={size}
                bottomSheet={bottomSheet}
              >
                <ModalContext.Provider
                  value={{ titleProps, title, close: onClose, showDismiss }}
                >
                  {children}
                  {isDismissable === true && (
                    <ModalCrossButton
                      size="S"
                      icon="24/Close"
                      onClick={onClose}
                    />
                  )}
                </ModalContext.Provider>
              </ModalDialog>
            </FocusScope>
          </ModalBackground>
        </Overlay>
      )
  )
}

const ModalContext = React.createContext<{
  titleProps: React.HTMLAttributes<HTMLElement>
  title: string
  close?: () => void
  showDismiss: boolean
}>({
  titleProps: {},
  title: '',
  close: undefined,
  showDismiss: true,
})

const ModalBackground = animated(styled.div<{ zIndex: number }>`
  z-index: ${({ zIndex }) => zIndex};
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;

  ${theme((o) => [o.bg.surface4])}
`)

const ModalDialog = animated(styled.div<{
  size: 'S' | 'M' | 'L'
  bottomSheet: boolean | 'full'
}>`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: ${(p) =>
    p.size === 'S'
      ? columnSystem(3, COLUMN_UNIT, GUTTER_UNIT) + GUTTER_UNIT * 2
      : p.size === 'M'
      ? columnSystem(4, COLUMN_UNIT, GUTTER_UNIT) + GUTTER_UNIT * 2
      : // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
      p.size === 'L'
      ? columnSystem(6, COLUMN_UNIT, GUTTER_UNIT) + GUTTER_UNIT * 2
      : unreachable(p.size)}px;

  ${theme((o) => [o.bg.background1, o.borderRadius(24)])}

  @media ${({ theme }) => maxWidth(theme.breakpoint.screen1)} {
    ${(p) =>
      p.bottomSheet === 'full'
        ? css`
            top: auto;
            bottom: 0;
            left: 0;
            transform: none;
            border-radius: 0;
            width: 100%;
            height: 100%;
          `
        : p.bottomSheet
        ? css`
            top: auto;
            bottom: 0;
            left: 0;
            transform: none;
            border-radius: 0;
            width: 100%;
          `
        : css`
            width: calc(100% - 48px);
          `}
  }
`)

const ModalCrossButton = styled(IconButton)`
  position: absolute;
  top: 8px;
  right: 8px;

  ${theme((o) => [o.font.text3.hover.press])}
`

export function ModalTitle(props: React.HTMLAttributes<HTMLHeadingElement>) {
  const { titleProps, title } = useContext(ModalContext)
  return (
    <ModalHeading {...titleProps} {...props}>
      {title}
    </ModalHeading>
  )
}

const ModalHeading = styled.h3`
  margin: 0;
  font-weight: inherit;
  font-size: inherit;
`

export function ModalDismissButton({ children, ...props }: ButtonProps) {
  const { close, showDismiss } = useContext(ModalContext)

  if (!showDismiss) {
    return null
  }

  return (
    <Button {...props} onClick={close} fixed>
      {children}
    </Button>
  )
}
