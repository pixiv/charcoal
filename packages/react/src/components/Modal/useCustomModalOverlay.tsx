import * as React from 'react'
import {
  AriaModalOverlayProps,
  ModalOverlayAria,
  ariaHideOutside,
  useOverlay,
  useOverlayFocusContain,
} from '@react-aria/overlays'
import { usePreventScroll } from '../DropdownSelector/Popover/usePreventScroll'

/**
 * We want to enable scrolling on the modal background,
 * but `useModalOverlay` (specifically, `useOverlay` within it) detects pointer events on the scrollbar.
 * Therefore, to prevent this issue, we need to override `shouldCloseOnInteractOutside` in `useModalOverlay`.
 */
export function useCharcoalModalOverlay(
  props: AriaModalOverlayProps,
  state: { isOpen: boolean; onClose: () => void },
  ref: React.RefObject<HTMLElement>
): ModalOverlayAria {
  const { overlayProps, underlayProps } = useOverlay(
    {
      ...props,
      isOpen: state.isOpen,
      onClose: state.onClose,
      shouldCloseOnInteractOutside: () => false,
    },
    ref
  )

  usePreventScroll(
    typeof document !== 'undefined' ? document.body : null,
    state.isOpen
  )

  useOverlayFocusContain()

  React.useEffect(() => {
    if (state.isOpen && ref.current) {
      return ariaHideOutside([ref.current])
    }
  }, [state.isOpen, ref])

  return {
    modalProps: overlayProps,
    underlayProps,
  }
}
