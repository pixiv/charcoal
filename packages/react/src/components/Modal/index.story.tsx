import React from 'react'
import { Story } from '../../_lib/compat'
import Modal, { ModalDismissButton, ModalProps } from '.'
import { OverlayProvider } from '@react-aria/overlays'
import { useOverlayTriggerState } from 'react-stately'
import Button from '../Button'
import {
  ModalAlign,
  ModalBody,
  ModalButtons,
  ModalHeader,
} from './ModalPlumbing'
import styled from 'styled-components'
import { theme } from '../../styled'
import TextField from '../TextField'

export default {
  title: 'Modal',
  component: Modal,
  args: {
    title: 'Title',
  },
  argTypes: {
    size: {
      options: ['S', 'M', 'L'],
      control: {
        type: 'inline-radio',
      },
    },
    bottomSheet: {
      options: ['full', 'true', 'false'],
      mapping: { full: 'full', true: true, false: false },
      control: {
        type: 'inline-radio',
      },
    },
  },
}

const DefaultStory = (args: ModalProps) => {
  const state = useOverlayTriggerState({})
  return (
    // Application must be wrapped in an OverlayProvider so that it can be
    // hidden from screen readers when a modal opens.
    <OverlayProvider>
      <Button onClick={() => state.open()}>Open Modal</Button>

      <Modal
        {...args}
        isOpen={state.isOpen}
        onClose={() => state.close()}
        isDismissable
      >
        <ModalHeader />
        <ModalBody>
          <ModalVStack>
            <StyledModalText>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Quod
              placeat tenetur, necessitatibus laudantium cumque exercitationem
              provident. Quaerat iure enim, eveniet dolores earum odio quo
              possimus fugiat aspernatur, numquam, commodi repellat.
            </StyledModalText>
            <ModalAlign>
              <TextField
                showLabel
                label="Name"
                placeholder="Nagisa"
              ></TextField>
            </ModalAlign>
            <ModalAlign>
              <TextField
                showLabel
                label="Country"
                placeholder="Tokyo"
              ></TextField>
            </ModalAlign>
          </ModalVStack>
          <ModalButtons>
            <Button variant="Primary" onClick={() => state.close()} fixed>
              Apply
            </Button>
            <Button onClick={() => state.close()} fixed>
              Cancel
            </Button>
          </ModalButtons>
        </ModalBody>
      </Modal>
    </OverlayProvider>
  )
}

const ModalVStack = styled.div`
  display: grid;
  gap: 24px;
`

const StyledModalText = styled(ModalAlign)`
  ${theme((o) => [o.font.text2, o.typography(14)])}
`

export const Default: Story<ModalProps> = DefaultStory.bind({})

const FullBottomSheetStory = (args: ModalProps) => {
  const state = useOverlayTriggerState({})
  return (
    // Application must be wrapped in an OverlayProvider so that it can be
    // hidden from screen readers when a modal opens.
    <OverlayProvider>
      <Button onClick={() => state.open()}>Open Modal</Button>

      <Modal
        {...args}
        isDismissable
        bottomSheet="full"
        isOpen={state.isOpen}
        onClose={() => state.close()}
      >
        <ModalHeader />
        <ModalBody>
          <ModalVStack>
            <StyledModalText>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Quod
              placeat tenetur, necessitatibus laudantium cumque exercitationem
              provident. Quaerat iure enim, eveniet dolores earum odio quo
              possimus fugiat aspernatur, numquam, commodi repellat.
            </StyledModalText>
            <ModalAlign>
              <TextField
                showLabel
                label="Name"
                placeholder="Nagisa"
              ></TextField>
            </ModalAlign>
            <ModalAlign>
              <TextField
                showLabel
                label="Country"
                placeholder="Tokyo"
              ></TextField>
            </ModalAlign>
          </ModalVStack>
          <ModalButtons>
            <Button variant="Primary" onClick={() => state.close()} fixed>
              Apply
            </Button>
            <Button onClick={() => state.close()} fixed>
              Cancel
            </Button>
          </ModalButtons>
        </ModalBody>
      </Modal>
    </OverlayProvider>
  )
}

export const FullBottomSheet: Story<ModalProps> = FullBottomSheetStory.bind({})

const BottomSheetStory = (args: ModalProps) => {
  const state = useOverlayTriggerState({})
  return (
    // Application must be wrapped in an OverlayProvider so that it can be
    // hidden from screen readers when a modal opens.
    <OverlayProvider>
      <Button onClick={() => state.open()}>Open Modal</Button>

      <Modal
        {...args}
        isOpen={state.isOpen}
        onClose={() => state.close()}
        bottomSheet
        isDismissable
      >
        <ModalHeader />
        <ModalBody>
          <ModalVStack>
            <StyledModalText>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Quod
              placeat tenetur, necessitatibus laudantium cumque exercitationem
              provident. Quaerat iure enim, eveniet dolores earum odio quo
              possimus fugiat aspernatur, numquam, commodi repellat.
            </StyledModalText>
          </ModalVStack>
          <ModalButtons>
            <Button variant="Danger" onClick={() => state.close()} fixed>
              削除する
            </Button>
            <ModalDismissButton>キャンセル</ModalDismissButton>
          </ModalButtons>
        </ModalBody>
      </Modal>
    </OverlayProvider>
  )
}

export const BottomSheet: Story<ModalProps> = BottomSheetStory.bind({})
