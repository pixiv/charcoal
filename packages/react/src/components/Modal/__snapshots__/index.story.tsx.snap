// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal BottomSheet smoke-test 1`] = `
.c0 {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  padding: 0;
  border-style: none;
  outline: none;
  color: inherit;
  text-rendering: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  word-spacing: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: inherit;
  margin: 0;
  overflow: visible;
  text-transform: none;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c1 {
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  display: inline-grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  border-radius: 999999px;
  font-size: 14px;
  line-height: 22px;
  font-weight: bold;
  padding-right: 24px;
  padding-left: 24px;
  color: var(--charcoal-text2);
  background-color: var(--charcoal-surface3);
  -webkit-transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  height: 40px;
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active,
.c1:not(:disabled):not([aria-disabled]):focus,
.c1[aria-disabled='false']:focus,
.c1:not(:disabled):not([aria-disabled]):focus-visible,
.c1[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c1:not(:disabled):not([aria-disabled]):hover,
.c1[aria-disabled='false']:hover {
  color: var(--charcoal-text2-hover);
  background-color: var(--charcoal-surface3-hover);
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active {
  color: var(--charcoal-text2-press);
  background-color: var(--charcoal-surface3-press);
}

.c1:disabled,
.c1[aria-disabled]:not([aria-disabled='false']) {
  opacity: 0.32;
}

<div data-dark="false">
  <div data-overlay-container="true">
    <button class="c0 c1">
      Open Modal
    </button>
  </div>
</div>
`;

exports[`Modal Default smoke-test 1`] = `
.c0 {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  padding: 0;
  border-style: none;
  outline: none;
  color: inherit;
  text-rendering: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  word-spacing: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: inherit;
  margin: 0;
  overflow: visible;
  text-transform: none;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c1 {
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  display: inline-grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  border-radius: 999999px;
  font-size: 14px;
  line-height: 22px;
  font-weight: bold;
  padding-right: 24px;
  padding-left: 24px;
  color: var(--charcoal-text2);
  background-color: var(--charcoal-surface3);
  -webkit-transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  height: 40px;
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active,
.c1:not(:disabled):not([aria-disabled]):focus,
.c1[aria-disabled='false']:focus,
.c1:not(:disabled):not([aria-disabled]):focus-visible,
.c1[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c1:not(:disabled):not([aria-disabled]):hover,
.c1[aria-disabled='false']:hover {
  color: var(--charcoal-text2-hover);
  background-color: var(--charcoal-surface3-hover);
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active {
  color: var(--charcoal-text2-press);
  background-color: var(--charcoal-surface3-press);
}

.c1:disabled,
.c1[aria-disabled]:not([aria-disabled='false']) {
  opacity: 0.32;
}

<div data-dark="false">
  <div data-overlay-container="true">
    <button class="c0 c1">
      Open Modal
    </button>
  </div>
</div>
`;

exports[`Modal FullBottomSheet smoke-test 1`] = `
.c0 {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  padding: 0;
  border-style: none;
  outline: none;
  color: inherit;
  text-rendering: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  word-spacing: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: inherit;
  margin: 0;
  overflow: visible;
  text-transform: none;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c1 {
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  display: inline-grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  border-radius: 999999px;
  font-size: 14px;
  line-height: 22px;
  font-weight: bold;
  padding-right: 24px;
  padding-left: 24px;
  color: var(--charcoal-text2);
  background-color: var(--charcoal-surface3);
  -webkit-transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  height: 40px;
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active,
.c1:not(:disabled):not([aria-disabled]):focus,
.c1[aria-disabled='false']:focus,
.c1:not(:disabled):not([aria-disabled]):focus-visible,
.c1[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c1:not(:disabled):not([aria-disabled]):hover,
.c1[aria-disabled='false']:hover {
  color: var(--charcoal-text2-hover);
  background-color: var(--charcoal-surface3-hover);
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active {
  color: var(--charcoal-text2-press);
  background-color: var(--charcoal-surface3-press);
}

.c1:disabled,
.c1[aria-disabled]:not([aria-disabled='false']) {
  opacity: 0.32;
}

<div data-dark="false">
  <div data-overlay-container="true">
    <button class="c0 c1">
      Open Modal
    </button>
  </div>
</div>
`;

exports[`Modal InternalScroll smoke-test 1`] = `
.c0 {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  padding: 0;
  border-style: none;
  outline: none;
  color: inherit;
  text-rendering: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  word-spacing: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: inherit;
  margin: 0;
  overflow: visible;
  text-transform: none;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c1 {
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  display: inline-grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  border-radius: 999999px;
  font-size: 14px;
  line-height: 22px;
  font-weight: bold;
  padding-right: 24px;
  padding-left: 24px;
  color: var(--charcoal-text2);
  background-color: var(--charcoal-surface3);
  -webkit-transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  height: 40px;
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active,
.c1:not(:disabled):not([aria-disabled]):focus,
.c1[aria-disabled='false']:focus,
.c1:not(:disabled):not([aria-disabled]):focus-visible,
.c1[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c1:not(:disabled):not([aria-disabled]):hover,
.c1[aria-disabled='false']:hover {
  color: var(--charcoal-text2-hover);
  background-color: var(--charcoal-surface3-hover);
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active {
  color: var(--charcoal-text2-press);
  background-color: var(--charcoal-surface3-press);
}

.c1:disabled,
.c1[aria-disabled]:not([aria-disabled='false']) {
  opacity: 0.32;
}

<div data-dark="false">
  <div data-overlay-container="true">
    <button class="c0 c1">
      Open Modal
    </button>
  </div>
</div>
`;

exports[`Modal NotDismmissableStory smoke-test 1`] = `
.c0 {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  padding: 0;
  border-style: none;
  outline: none;
  color: inherit;
  text-rendering: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  word-spacing: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font: inherit;
  margin: 0;
  overflow: visible;
  text-transform: none;
}

.c0:disabled,
.c0[aria-disabled]:not([aria-disabled=false]) {
  cursor: default;
}

.c0:focus {
  outline: none;
}

.c0::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c1 {
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  display: inline-grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  border-radius: 999999px;
  font-size: 14px;
  line-height: 22px;
  font-weight: bold;
  padding-right: 24px;
  padding-left: 24px;
  color: var(--charcoal-text2);
  background-color: var(--charcoal-surface3);
  -webkit-transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  transition: 0.2s color,0.2s background-color,0.2s box-shadow;
  height: 40px;
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active,
.c1:not(:disabled):not([aria-disabled]):focus,
.c1[aria-disabled='false']:focus,
.c1:not(:disabled):not([aria-disabled]):focus-visible,
.c1[aria-disabled='false']:focus-visible {
  outline: none;
  box-shadow: 0 0 0 4px rgba(0,150,250,0.32);
}

.c1:not(:disabled):not([aria-disabled]):hover,
.c1[aria-disabled='false']:hover {
  color: var(--charcoal-text2-hover);
  background-color: var(--charcoal-surface3-hover);
}

.c1:not(:disabled):not([aria-disabled]):active,
.c1[aria-disabled='false']:active {
  color: var(--charcoal-text2-press);
  background-color: var(--charcoal-surface3-press);
}

.c1:disabled,
.c1[aria-disabled]:not([aria-disabled='false']) {
  opacity: 0.32;
}

<div data-dark="false">
  <div data-overlay-container="true">
    <button class="c0 c1">
      Open Modal
    </button>
  </div>
</div>
`;
